### DISCUSSION TITLE
reload tag + db.add( ) causes "db-journal.db" to be created and intermittent "Failed to Fetch" errors in localhost:any-port/docs, no error in terminal

### DISCUSSION THREAD

**Discussion by joshb019919** [fastapi/fastapi#9022] [Questions]
### PREREQUISITES ###
* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

* [x] After submitting this, I commit to:
    * Get good enough to be able to help people when I can
    * I'm sure it'll be more than 2

Alright, so this contains a good deal of minimal code, but I can't see anything else to get rid of.

### SYSTEM/VERSIONS ###
Windows _7 x64 64-bit_
Chrome _83_
FastAPI, Pydantic, SQLAlchemy, Uvicorn _latest_
VSCode _latest_

### CODE | BUG ###
    undocumented | Failed to Fetch

QUIRKS:
    * works intermittently
    * causes multiple refreshes in a row
    * sometimes creates `db-journal.db`, and sometimes not

POSSIBLE REASONS GIVEN:
    * CORS
    * Network Failure
    * URL scheme must be `http` or `https` for CORS reques

EXCEPT FOR THESE FACTS:
    * HTTP protocols
    * Other websites have no issues
    * CORS is set

### PROBLEM LINES ###

    db.add(stock); lines 70 & 86
    --reload tag in terminal

Been debugging for a week+to try and fix it before I realized it was the `--reload` tag.  It's annoying to have to quit and rerun uvicorn each save, but it works.

I read somewhere that it has something to do with refreshing a database connection, but I don't know why it would fail.

### MINIMAL CODE ###

```Python
    from fastapi import BackgroundTasks, Depends, FastAPI
    from pydantic import BaseModel
    from sqlalchemy import Column, create_engine, DateTime, Integer, String
    from sqlalchemy.ext.declarative import declarative_base
    from sqlalchemy.orm import Session, sessionmaker
    from starlette.middleware.cors import CORSMiddleware
    
    # Standard for SQLite
    SQLALCHEMY_DATABASE_URL = "sqlite:///test.db"
    
    engine = create_engine(SQLALCHEMY_DATABASE_URL, 
                           connect_args={"check_same_thread": False})
    
    SessionLocal = sessionmaker(autocommit=False, 
                                autoflush=False, 
                                bind=engine)
    
    Base = declarative_base()
    
    
    # SQLAlchemy mapped model
    class Stock(Base):
        __tablename__ = "stocks"
    
        id = Column(Integer(), primary_key=True, index=True)
        datetime = Column(DateTime(), index=True)
        symbol = Column(String(5), nullable=False, index=True)
        company = Column(String(50), index=True)
        price = Column(Integer())
    
        def __repr__(self):
            return (f"""<Stock(id={self.id}, 
                               datetime={self.datetime}, 
                               company={self.company}, 
                               symbol={self.symbol}, 
                               price={self.price})>\n""")
    
    
    # Pydantic incoming stock model
    class StockRequest(BaseModel):
        symbol: str
    
    
    # Create the app, database, and stocks table
    app = FastAPI()
    Base.metadata.create_all(bind=engine)
    
    app.add_middleware(
        CORSMiddleware,
        allow_origins=["*"],
        allow_credentials=True,
        allow_methods=["*"],
        allow_headers=["*"],
    )
    
    
    # Fetch a fake stock quote 
    def fetch_stock(symbol: str):
        # Dummy db connection
        db = SessionLocal()
    
        # Fake quote
        quote = {
            "quote": {
                "companyName": "Ford Motor Co.",
                "symbol": "F",
                "price": 7.898}}
    
        # Submit for later query
        db.add(quote)
        db.commit()
        
        return quote
        
    
    @app.post("/stock/")
    async def create_stock(stock_request: StockRequest):
        # Dummy db connection
        db = SessionLocal()
        
        # Instantiate a Stock object and assign its symbol
        stock = Stock()
        stock.symbol = stock_request.symbol
    
        # Without this add/commit, the background task fails
        db.add(stock)
        db.commit()
        
        fetch_stock("F")
    
        # Query stocks table and print results
        query = db.query(Stock).all()
        for d in query:
            print(f"""{d.company}\t
                      {d.price}\t
                      {d.symbol}\t
                      {d.datetime}""")
    
        # If at all successful, returns query list
        return {
            "query": query
        }

**Comment by joshincr** [comment]
I am also facing the same issue while dealing with databases. @joshb019919 Any solution found to this other than having to reload uvicorn manually?

**Comment by joshb019919** [comment]
No, I never could get it to work, so I switched back to Flask.

**Comment by joshincr** [comment]
Okay. @tiangolo and team please look into this issue.

![Screenshot (330)](https://user-images.githubusercontent.com/78969613/122970764-ea609a80-d3ab-11eb-868a-896e44cf7fde.png)

**Comment by DjangodevArif** [comment]
I also face this issue ,I think it's happen when post or put request is called.
